pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -534,6 +535,10 @@ func (c *Reconciler) checkTimeout(tr *v1alpha1.TaskRun, ts *v1alpha1.TaskSpec, d
 	}
 
 	timeout := tr.Spec.Timeout.Duration
+	// If timeout is set to 0 or defaulted to 0, there is no timeout.
+	if timeout == apisconfig.NoTimeoutDuration {
+		return false, nil
+	}
```

Comment:
i think it would be worth adding some test coverage for this! (it's probably hard to do since `checkTimeout` is a function on the reconciler - probably we should move `checkTimeout` out into another package!)